libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1008,6 +1006,9 @@ class MapboxNavigation(
 
     private fun createInternalRoutesObserver() = RoutesObserver { routes ->
         tripSession.setRoute(routes.firstOrNull(), directionsSession.initialLegIndex)
+        if (routes.isNotEmpty()) {
+            routeRefreshController.restart(routes.first())
+        }
```

Comment:
When it is empty it should stop the previous route refresh right?

Reply:
Yes, good catch!